src/metricscollector/config/config.go

```diff
@@ -54,12 +55,21 @@ var defaultCollectorConfig = CollectorConfig{
 	PollInterval:    DefaultPollInterval,
 }
 
+type DataPruneConfig struct {
+	CutoffDays int `yaml:"cutoff_days"`
+}
+
+var defaultDataPruneConig = DataPruneConfig{
```

Comment:
you are duplicating yourself with your naming. You are already in the pruner package so metriccollector or whatever db table name .go is sufficient, pruner is redundant.
And these things are doing nothing more than driving the underlying DB code or in the case of scheduler, making the remote call.